src/oops/SOLID/openClosePrinciple/before/factory/TaxCalculatorFactory.java

```diff
@@ -0,0 +1,29 @@
+package oops.SOLID.openClosePrinciple.before.factory;
+
+import oops.SOLID.openClosePrinciple.before.employees.Employee;
+import oops.SOLID.openClosePrinciple.before.taxes.FullTimeEmployeeTaxCalculator;
+import oops.SOLID.openClosePrinciple.before.taxes.InternTaxCalculator;
+import oops.SOLID.openClosePrinciple.before.taxes.PartTimeEmployeeTaxCalculator;
+import oops.SOLID.openClosePrinciple.before.taxes.TaxScheme;
+
+public class TaxCalculatorFactory {
+
+    private final  String FULL_TIME_EMPLOYEE  = "FullTimeEmployee";
+    private final  String PART_TIME_EMPLOYEE  = "PartTimeEmployee";
+    private final   String INTERN = "Intern";
+    public TaxScheme getScheme(Employee employee){
+         switch (employee.getClass().getSimpleName()){
+             case FULL_TIME_EMPLOYEE:
+                 return new FullTimeEmployeeTaxCalculator();
```

Comment:
Just a small suggestion here, every time this method is called, a new object of type `TaxScheme` will get created. This will impact performance if number of times this method is getting called is quite high. I would be advisable if these instances are somehow cached and served on request.

See `prototype` pattern for more details.